llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/Legality.h

```diff
@@ -48,14 +106,37 @@ class Widen final : public LegalityResult {
   }
 };
 
+class Pack final : public LegalityResultWithReason {
+  Pack(ResultReason Reason)
+      : LegalityResultWithReason(LegalityResultID::Pack, Reason) {}
+  friend class LegalityAnalysis; // For constructor.
+
+public:
+  static bool classof(const LegalityResult *From) {
+    return From->getSubclassID() == LegalityResultID::Pack;
+  }
+};
+
 /// Performs the legality analysis and returns a LegalityResult object.
 class LegalityAnalysis {
+  /// Owns the legality result objects created by createLegalityResult().
+  SmallVector<std::unique_ptr<LegalityResult>> ResultPool;
+  /// Checks opcodes, types and other IR-specifics and returns nullopt if it is
+  /// not illegal to vectorize, otherwise it returns a ResultReason object.
+  std::optional<ResultReason>
+  cantVectorizeBasedOnOpcodesAndTypes(ArrayRef<Value *> Bndl);
```

Comment:
How does notVectorizableBasedOpcodesAndTypes sound?

Reply:
Yeah, it's easy to miss the "t" in "cant". I will rename it.